src/App/Cart/Service/CartCalculationService.php

```diff
@@ -80,4 +83,17 @@ protected function hasDeliveryOptions(PdkCart $cart): bool
 
         return $anyItemIsDeliverable && ! $deliveryOptionsDisabled;
     }
+
+    /**
+     * @param  \MyParcelNL\Pdk\App\Cart\Model\PdkCart $cart
+     * @param  string                                 $packageType
+     *
+     * @return bool
+     */
+    private function isWeightUnderPackageTypeLimit(PdkCart $cart, string $packageType): bool
+    {
+        return $cart->lines->reduce(static function ($carry, $line) use ($packageType) {
+                return $carry + $line->quantity * $line->product->weight;
+            }, 0) <= Pdk::get(sprintf('%sWeightMaximum', $packageType));
```

Comment:
Om dit netter te maken kan je denk ik beter die variabele in de config een array maken:

```
	`packageTypeWeightLimits' => value([
			DeliveryOptions::PACKAGE_TYPE_MAILBOX_NAME => 2000,
			// ... etc
	]),
```

En dan `Arr::get(Pdk::get('packageTypeWeightLimits'), $packageType))`. Dan hoeven ze niet eens allemaal te bestaan, dan krijgt hij een auto fallback naar null. (Die je zou kunnen aanpassen via de 2e param van `Arr::get`)

En nóg mooier is als je die reduce functie toevoegt aan de `PdkOrderLineCollection` als `getTotalWeight` oid.